radix-engine/src/vm/wasm_runtime/scrypto_runtime.rs

```diff
@@ -658,4 +683,50 @@ impl<'y, Y: SystemApi<RuntimeError>> WasmRuntime for ScryptoRuntime<'y, Y> {
 
         self.allocate_buffer(hash.to_vec())
     }
+
+    #[trace_resources(log=data.len())]
+    fn crypto_utils_blake2b_256_hash(
+        &mut self,
+        data: Vec<u8>,
+    ) -> Result<Buffer, InvokeError<WasmRuntimeError>> {
+        // TODO: costing
+
+        let hash = blake2b_256_hash(data);
+
+        self.allocate_buffer(hash.to_vec())
+    }
+
+    #[trace_resources(log=message.len())]
+    fn crypto_utils_ed25519_verify(
+        &mut self,
+        message: Vec<u8>,
+        public_key: Vec<u8>,
+        signature: Vec<u8>,
+    ) -> Result<u32, InvokeError<WasmRuntimeError>> {
+        let public_key: Ed25519PublicKey =
+            scrypto_decode(&public_key).map_err(WasmRuntimeError::InvalidPublicKey)?;
+        let signature: Ed25519Signature =
+            scrypto_decode(&signature).map_err(WasmRuntimeError::InvalidSignature)?;
+        // TODO: costing
+
+        Ok(verify_ed25519(&message, &public_key, &signature) as u32)
+    }
+
+    #[trace_resources(log=message.len())]
+    fn crypto_utils_secp256k1_ecdsa_verify(
+        &mut self,
+        message: Vec<u8>,
+        public_key: Vec<u8>,
+        signature: Vec<u8>,
+    ) -> Result<u32, InvokeError<WasmRuntimeError>> {
+        let public_key: Secp256k1PublicKey =
+            scrypto_decode(&public_key).map_err(WasmRuntimeError::InvalidPublicKey)?;
+        let signature: Secp256k1Signature =
+            scrypto_decode(&signature).map_err(WasmRuntimeError::InvalidSignature)?;
+        let hash = Hash::try_from(message.as_slice()).map_err(WasmRuntimeError::InvalidHash)?;
```

Comment:
Not sure, maybe better to panic here. 
In scrypto the respective method [accepts Hash](https://github.com/radixdlt/radixdlt-scrypto/pull/1942/files#diff-30648a31feb5946691c27f312d1ce916bea3b855a5dab7f33d2f6e84b4e42c1dR130).
If we panic, then no need to add this `WasmRuntimeError::InvalidHash` error and update  schema.

Reply:
On the other hand we can say that the same goes to public key and signature, the respective methods in Scrypto accept a key type or signature type, but we error handle it here anyway.
So maybe for consistency it is good to return `InvalidHash` here

Reply:
We should not panic here. Just because the scrypto message accepts a hash doesn't mean a WASM file couldn't use the low level API and circumvent that.

It's okay to add a new error variant though imo :+1:

Reply:
Done